### DISCUSSION TITLE
How to reset all in memory cache in logout handler?

### DISCUSSION THREAD

**Discussion by siamahnaf** [urql-graphql/urql#3210] [Q&A] [answered]
Hello, I want to reset or remove all in memory cache when my run logout handler in nextjs application.

My `logout` component-

```
useEffect(() => {
    if (logoutData.data?.logout.success) {
        router.push("/login", undefined, { shallow: true });
    }
}, [logoutData.data])
```

Here I run a `useEffect` hook to check logout success. When logout success is true, it redirect my page to login page.

But problem is, in urql cache all previous data stored in the cache. Here I should remove or reset all in memory cache. Is there any to to do that?

Here is my `client.ts`-

```
import { cacheExchange, Client, fetchExchange, ssrExchange, errorExchange } from "urql";
import { useMemo } from "react";

let client: Client | null = null;

let ssrCache: ReturnType<typeof ssrExchange> | null = null;

const isServer = typeof window === "undefined";
export function initUrqlClient(initialState?: any) {
    if (!client) {
        ssrCache = ssrExchange({ initialState, isClient: !isServer });
        client = new Client({
            url: process.env.NEXT_PUBLIC_API_URL as string,
            exchanges: [
                errorExchange({
                    onError: (error) => {
                        error.message = error.message.replace(/^\[.*\]\s*/, "");
                    }
                }),
                cacheExchange,
                ssrCache,
                fetchExchange,
            ],
            fetchOptions: {
                credentials: "include"
            },
        });
    } else {
        ssrCache?.restoreData(initialState);
    }
    return { client, ssrCache };
}


export const useClient = (pageProps?: any) => {
    const urqlData = pageProps.urqlState;
    const { client } = useMemo(() => {
        return initUrqlClient(urqlData);
    }, [urqlData]);
    return client;
};
```

**Comment by kitten** [comment, marked as answer]
Hiya,

there are some other issues and discussions for this, I believe, so a quick search should bring those up for you.

The short idea here is, if you'd like to reset the cache, re-create the `Client`. The recommended way is to swap out the `Client` becaus it'll erase all state in all exchanges, not just the cache.

**Reply by siamahnaf** [reply]
First of all, I already search for the same things. But I am in darkness, because I hope there is different way to reset cache.

Secondly, thank you for your idea. As you say, I need to recreate the `Client'. Okay now. Thanks.
